src/term.c

```diff
@@ -479,12 +499,22 @@ static int cmd_write(PROGRAMMER *pgm, AVRPART *p, int argc, char *argv[]) {
   if (strcmp(argv[argc - 1], "...") == 0) {
     write_mode = WRITE_MODE_FILL;
     start_offset = 4;
-    len = strtoul(argv[3], &end_ptr, 0);
+    len = strtol(argv[3], &end_ptr, 0);
     if (*end_ptr || (end_ptr == argv[3])) {
       pmsg_error("(write ...) cannot parse length %s\n", argv[3]);
       free(buf);
       return -1;
     }
+    // Turn negative len value (no. bytes from top of memory) into an actual length number
+    if (len < 0)
+      len = maxsize + len - addr + 1;
+    if (len == 0)
+      return 0;
+    if (len < 0 || len > maxsize - addr) {
+      pmsg_error("(write) effective %s start address 0x%0*x and effective length %d not compatible with memory size %d\n",
+        mem->desc, maxsize > 0x10000? 5: 4, addr, len, maxsize);
+      return -1;
```

Comment:
Good to mention which memory. I thought mentioning (write ...) not (write) is better (as it's the write fill mode here)